components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerImpl.java

```diff
@@ -224,20 +315,20 @@ private void validateSecretDeleteRequest(String secretName)
     }
 
     /**
-     * Validate that secret name and value are non-empty.
+     * Validate that secret type and secret name and value are non-empty.
      * Set tenant domain if they are not set to the secret object.
      *
      * @param secret The secret to be added.
      * @throws SecretManagementException If secret validation fails.
      */
-    private void validateSecretCreateRequest(Secret secret) throws SecretManagementException {
+    private void validateSecretCreateRequest(String secretTypeName, Secret secret) throws SecretManagementException {
 
-        if (StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
+        if (StringUtils.isEmpty(secretTypeName) || StringUtils.isEmpty(secret.getSecretName()) || StringUtils.isEmpty(secret.getSecretValue())) {
```

Comment:
Seems like this has exceeded 120 character limit

Reply:
Addressed with commit [001ee25](https://github.com/wso2/carbon-identity-framework/pull/3762/commits/001ee2501f3eab862efd07a13864e81939cfe648)